src/library/coverartutils.cpp

```diff
@@ -84,34 +84,35 @@ QImage CoverArtUtils::loadCover(const CoverInfo& info) {
 }
 
 //static
-CoverArt CoverArtUtils::guessCoverArt(TrackPointer pTrack) {
-    CoverArt art;
-    art.info.source = CoverInfo::GUESSED;
-
-    if (pTrack.isNull()) {
-        return art;
+CoverInfo CoverArtUtils::guessCoverInfo(const Track* pTrack) {
```

Comment:
Why use a pointer instead of a const-ref? The track object is not modified, as far as I see. The caller should decide what happens if pTrack is a nullptr. In this case invoking this function does not make sense at all.